src/Microsoft.Data.Sqlite.Core/SqliteDataReader.cs

```diff
@@ -153,6 +150,7 @@ public override bool NextResult()
             raw.sqlite3_reset(_stmt);
 
             (_stmt, _hasRows) = _stmtQueue.Dequeue();
+            _record = new SqliteDataRecord(_stmt);
```

Comment:
Have you profiled the perf impact of this? My concern is that this is the read hot path, and this adds an allocation per row, a potentially huge increase in memory traffic. Unless I missed it, SqliteDataRecord is only used from SqliteDataReader, which means it could stay inside this type and doesn't need to be factored out into a separate type.

Reply:
We've never profiled this product. We should.

This is only one allocation per result set (typically one per reader), not row. It shouldn't be significant.

No multiple inheritance; composition was the only option.

Reply:
> This is only one allocation per result set

Oh right. NextResult() means next result of a statement, not next row. Ok, my concern is gone.